qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -521,10 +577,17 @@ def backends(
                 ):
                     backends.append(backend)
         else:
-            if instance:
-                raise IBMInputValueError(
-                    "The 'instance' keyword is only supported for ``ibm_quantum`` runtime."
-                )
+            if self._client_params.instance != self._account.instance:
+                # reset api client
+                self._client_params.instance = self._account.instance
+                self._api_client = RuntimeClient(self._client_params)
+            # do we want backends from all instances in service.backends()?
+            if not self._all_instances:
+                # fetch all instances and all backends in each instance. Save in dict
+                self._all_instances = self._account.list_instances()  # type: ignore[attr-defined]
+                for inst in self._all_instances:
+                    self._discover_backends_from_instance(inst)
+
```

Comment:
I think that it would make sense to only fetch backends from all instances if no `instance` or `default_instance` have been set.  The code using `list_instances()` doesn't seem to be doing this as far as I can see. Iterating over instances will make fetching backends much slower so users should have a way of calling `backends()` without this additional overhead.